src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/AttachedCollections/IDependenciesTreeProjectSearchContext.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Threading;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Tree.Dependencies.AttachedCollections
+{
+    /// <summary>
+    /// Provides services throughout the lifetime of a search operation.
+    /// </summary>
+    public interface IDependenciesTreeProjectSearchContext
+    {
+        /// <summary>
+        /// Gets a token that aborts the search operation if cancelled.
+        /// </summary>
+        CancellationToken CancellationToken { get; }
+
+        /// <summary>
+        /// Gets the unconfigured project being searched.
+        /// </summary>
+        UnconfiguredProject UnconfiguredProject { get; }
+
+        /// <summary>
+        /// The set of target frameworks exposed by this project.
+        /// </summary>
+        ImmutableArray<string> TargetFrameworks { get; }
```

Comment:
We should avoid exposing TargetFrameworks, and instead be using CPS concepts; Configurations, Dimensions Names and Values, ConfigurationGroups, etc.